docs/DTW_MP_trial.ipynb

```diff
@@ -0,0 +1,810 @@
+{
```

Comment:
<p>So, the slowness likely comes in these three lines:</p><p> </p><pre class="ql-syntax" spellcheck="false">a&nbsp;= (T_A[uint64_i:uint64_i+uint64_m] - μ_Q[uint64_i])*σ_Q_inverse[uint64_i]
b = (T_B[uint64_j:uint64_j+uint64_m] - M_T[uint64_j])*Σ_T_inverse[uint64_j]
dist = _njit_dtw(a, b, r)# ------------ update
</pre><p> </p><p>because you are not benefitting from traversing the diagonal and re-using the last covariance. Instead, you are computing the individual variances from scratch during each iteration and this is computationally very expensive.</p><p> </p><p>I think what we need to do is express the variance in a "rolling window" approach. See <code>core._welford_nanvar</code> for some possible inspiration </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/TDAmeritrade/stumpy/pull/733/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>